test/Npgsql.Tests/PoolTests.cs

```diff
@@ -7,14 +7,14 @@
 
 namespace Npgsql.Tests
 {
-    [NonParallelizable]
```

Comment:
I see you've added ApplicationName to all tests here for pool isolation. While that's a good thing, for tests that do this we should also clear the pool in a finally clause at the test's end, otherwise the physical connection created in that specific pool will live forever during the test run (and could ultimately cause connection exhaustion).

Also, I'm not sure it's a good idea to remove [NonParallelizable]. I don't mind if our tests take a few seconds longer just to be more sure that there's no multi-threading issues, because someone forgot an ApplicationName somewhere or similar... Note also that we have PoolManager which really is a shared global/static component, and even though we currently may not have tests exercising it specifically we may add some at some point.

Reply:
Shouldn't idle pruning take care of that? I kid I kid. Good point, I haven't seen it being done consistently (or at all) really, but I'll see if I can wrap that into something easily reusable for all tests in this class.

To your remark to remove NonParallelizable, I'm neutral about it but I don't see a lot of negatives, ultimately your call!

If pool manager would have tests at some point, wouldn't they be on a different class that could then have NonParallelizable again? It seems like a non issue for this.

Reply:
You're probably right about pool manager, but I'd rather we kept NonParallelizable just to be on the safe side. There is nothing in the world that's more annoying then test multithreading/isolation issues and we gain very little parallelizing this test suite...